### PR TITLE
add redis to rate limiting service

### PR DESCRIPTION
**Pull request by briangallagher**
# Description

## How to Verify

- Install manged-api and verify it's as expected 

- Verify that the rate-limiting Deployment exists
`oc get deployments/ratelimit -n redhat-rhmi-marin3r`

- Verify redis installed
`oc get redis ratelimit-service-redis-managed-api`

- Check the Pod logs of rate-limiting service and confirm no error when it connects to redis.

### CODE REVIEW

File: pkg/products/marin3r/reconciler.go [hunk 1]
```diff
@@ -220,6 +295,10 @@ func (r *Reconciler) preUpgradeBackupExecutor() backup.BackupExecutor {
 	if r.installation.Spec.UseClusterStorage != "false" {
 		return backup.NewNoopBackupExecutor()
 	}
-	//todo add backup for redis once it's added to the reconciler
-	return backup.NewNoopBackupExecutor()
+
+	return backup.NewAWSBackupExecutor(
+		r.installation.Namespace,
+		externalRedisSecretName,
```
Comment (sergioifg94) [inline pkg/products/marin3r/reconciler.go:301]:
```suggestion
		fmt.Sprintf("%s%s", constants.RateLimitRedisPrefix, r.installation.Name),
```

This parameter is the name of the Redis CR, which is referenced by the RedisSnapshot created when running the backups

**Review by sergioifg94 — CHANGES_REQUESTED** [review]
Verified on the cluster, and the code looks good. Only one small suggestion

File: pkg/products/marin3r/reconciler.go (hunk 1 above)
Comment (briangallagher) [reply pkg/products/marin3r/reconciler.go]: Good catch, thanks

**Comment by sergioifg94** [comment]
I'm not sure if it's ready to merge given that it's marked as WIP but it looks good to me
/lgtm

**Pull request merged**